sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/blob/perf/DownloadBlobNonSharedClientTest.java

```diff
@@ -13,44 +13,52 @@
 import java.io.OutputStream;
 
 public class DownloadBlobNonSharedClientTest extends AbstractDownloadTest<BlobPerfStressOptions> {
-    private static final int BUFFER_SIZE = 16 * 1024 * 1024;
-    private static final OutputStream DEV_NULL = new NullOutputStream();
+    private static final int DEFAULT_BUFFER_SIZE = 16 * 1024 * 1024;
     String blobName = "downloadTest";
 
-    private final byte[] buffer = new byte[BUFFER_SIZE];
+    private final OutputStream devNull = new NullOutputStream();
+
+    private final int bufferSize;
+    private final byte[] buffer;
 
     public DownloadBlobNonSharedClientTest(BlobPerfStressOptions options) {
         super(options);
+        // Dynamically determine the buffer size to be the minimum of 2 * the download size or 16MB.
+        // This reduces heap allocations when running smaller tests.
+        this.bufferSize = (int) Math.min(2 * options.getSize(), DEFAULT_BUFFER_SIZE);
```

Comment:
```suggestion
        this.bufferSize = (int) Math.min(options.getSize(), DEFAULT_BUFFER_SIZE);
```

As discussed offline, I don't see any reason to use `2 * size` rather than just `size`.